pkg/virt-controller/watch/snapshot/source.go

```diff
@@ -47,8 +51,10 @@ type snapshotSource interface {
 	Frozen() (bool, error)
 	Freeze() error
 	Unfreeze() error
-	Spec() snapshotv1.SourceSpec
-	PersistentVolumeClaims() map[string]string
+	HaveHotplug() (bool, error)
+	GetVMRevision() (*kubevirtv1.VirtualMachine, error)
```

Comment:
Should `VM` be in the name?  We eventually want to snapshot VMIs

Reply:
Also, if this function is not called by the controller directly it probably shouldn't be part of the interface

Reply:
even when we will want to snapshot VMIs we said that we will need this vm revision no?

Reply:
There can be VMIs without VMs

Reply:
then what should be done in case there is no VM and no vm revision?